### ISSUE TITLE
Can't get `ElementRef` type from `styled` comp

### ISSUE THREAD

**Issue by jjenzz** [nandorojo/dripsy#225]
The following `CardElement` type will be `never` but ideally would be `ReactNative.View`.

```jsx
const Card = styled(ReactNative.View)({});

// Later 
type CardElement = React.ElementRef<typeof Card>;
```

Here's an example that works as expected without the `styled` wrapper:

![image](https://user-images.githubusercontent.com/175330/183400472-d9782920-9880-42d0-9037-ad36618d7f38.png)

**Comment by jjenzz** [comment]
i wanted to have a quick look at solving this but i'm having trouble getting the project to run locally. 

i need to run `yarn` before i can do `yarn bootstrap` (from the contributing guidelines) but it fails at the `postinstall` step:

![image](https://user-images.githubusercontent.com/175330/183403449-9e4eba5b-ef08-4f4c-be9f-16ab12e87be4.png)

if you know a quick solution to this, i'm happy to take a look at the types :)

**Comment by nandorojo** [comment]
Hey @jjenzz, sorry I missed this. Happy to help you get this spun up.

> if you know a quick solution to this, i'm happy to take a look at the types :)

Could you suggest it here? 

I'm also going to clone this fresh to try to reproduce the setup issue you got. I was first getting used to complex types when TS released 4.4 with string literals, so the types for Dripsy are a bit of a mess haha. I've been wanting to get `ElementRef` working for a while, so if you have a solution that would be amazing. I've always found it hard to get it working when using an HOC wrapping a component which _might_ have a ref.

**Comment by jjenzz** [comment]
i've since gotten a new laptop and pulling the project worked fine this time. i've pushed a pr with a fix https://github.com/nandorojo/dripsy/pull/237
